p4-16/psa/PSA.mdk

```diff
@@ -814,3 +814,154 @@ only occurs once in the source code for control block `C`.  Some
 semi-precise definition of "after in-line expansion of all control
 block apply calls, and all action calls nested inside of other action
 calls" might do it.
+
+
+The tables below summarize restrictions that a PSA portable program
+must obey to maximize portability.  PSA compliant implementations are
+allowed to reject programs that attempt to do anything not allowed
+here.
+
+Some of these are limitations of the P4_16 language itself, but some
+are unique to the PSA architecture.
+
+The stateful types of objects in P4_16 are packages, parsers,
+controls, externs, and tables.  All other types are referred to as
+"value types" here.
+
+Some guiding principles:
+
+- Packages can be instantiated, but have no run-time behavior, so
+  there is no use in passing them to anything else other than another
+  package.
+- Controls are not allowed to call parsers in a PSA portable program,
+  and vice versa, so there is no use in passing one type to the other
+  in constructor parameters or run-time parameters.
+- Externs are not allowed to call parsers or controls (at least in
+  PSA, and probably also P4_16 as well), so there is no use in passing
+  objects of those types to them.
+- Tables are always instantiated directly in their enclosing control,
+  and cannot be instantiated at the top level.  There is no type by
+  which a table can be passed as either a constructor or run-time
+  parameter anywhere else.  They are only intended to be used from
+  within the control where they are defined.
+
+The first table lists restrictions on what types can be passed as
+constructor parameters to other types.
+
+|-------------|-------------------------------------------------|
```

Comment:
Regarding moving these to the P4_14 spec instead, I have no objections to doing so.  It seems worth asking the P4_16 language working group members to review it first.